lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -131,18 +144,15 @@ public ClassLoader getClassLoader() {
 
   /**
    * Registers the given conversion to be used when reading and writing with this
-   * data model.
+   * data model. Conversions can also be registered automatically, as documented
+   * on the class {@link Conversion Conversion&lt;T&gt;}.
    *
    * @param conversion a logical type Conversion.
    */
   public void addLogicalTypeConversion(Conversion<?> conversion) {
     conversions.put(conversion.getLogicalTypeName(), conversion);
     Class<?> type = conversion.getConvertedType();
-    Map<String, Conversion<?>> conversions = conversionsByClass.get(type);
-    if (conversions == null) {
-      conversions = new LinkedHashMap<>();
-      conversionsByClass.put(type, conversions);
-    }
+    Map<String, Conversion<?>> conversions = conversionsByClass.computeIfAbsent(type, k -> new LinkedHashMap<>());
```

Comment:
maybe rename this local variable to `conversionsByType` to avoid confusion with the field with the same name ?

Reply:
Agreed!